lib/optimizely/decision_service.rb

```diff
@@ -239,50 +220,46 @@ def get_variation_for_feature_rollout(feature_flag, user_id, attributes = nil)
 
       rollout = @config.get_rollout_from_id(rollout_id)
       unless rollout.nil? or rollout['experiments'].empty?
-        rollout_experiments = rollout['experiments']
-        number_of_rules = rollout_experiments.length - 1
+        rollout_rules = rollout['experiments']
+        number_of_rules = rollout_rules.length - 1
 
         # Go through each experiment in order and try to get the variation for the user
         for index in (0...number_of_rules)
-          experiment = rollout_experiments[index]
-          experiment_key = experiment['key']
+          rollout_rule = rollout_rules[index]
+          experiment_key = rollout_rule['key']
 
           # Check that user meets audience conditions for targeting rule
-          unless Audience.user_in_experiment?(@config, experiment, attributes)
+          unless Audience.user_in_experiment?(@config, rollout_rule, attributes)
             @config.logger.log(
               Logger::DEBUG,
-              "User '#{user_id}' does not meet the conditions to be in experiment '#{experiment_key}'."
+              "User '#{user_id}' does not meet the audience conditions to be in rollout rule '#{experiment_key}'."
             )
             # move onto the next targeting rule
             next
           end
 
           @config.logger.log(
             Logger::DEBUG,
-            "User '#{user_id}' meets conditions for targeting rule '#{index + 1}'."
+            "Attempting to bucket user '#{user_id}' into rollout rule '#{experiment_key}'."
           )
-          variation = @bucketer.bucket(experiment, user_id)
+          # Evaluate if user satisfies the traffic allocation for this rollout rule
+          variation = @bucketer.bucket(rollout_rule, user_id)
           unless variation.nil?
-            variation_key = variation['key']
             return variation
           end
 
           # User failed traffic allocation, jump to Everyone Else rule
           @config.logger.log(
             Logger::DEBUG,
-            "User '#{user_id}' is not in the traffic group for the targeting rule. Checking 'Eveyrone Else' rule now."
+            "User '#{user_id}' was excluded due to traffic allocation. Checking 'Eveyrone Else' rule now."
           )
           break
         end
 
-        # Evalute the "Everyone Else" rule, which is the last rule.
-        everyone_else_experiment = rollout_experiments[number_of_rules]
+        # get last rule which is the everyone else rule
+        everyone_else_experiment = rollout_rules[number_of_rules]
         variation = @bucketer.bucket(everyone_else_experiment, user_id)
         unless variation.nil?
```

Comment:
Please add 
"User '{$userId}'  was excluded from the 'Everyone Else' rule for feature flag" when variation is nil. 
See PHP/Java